lib/game_teams_manager.rb

```diff
@@ -12,22 +12,46 @@ def initialize(game_teams_path)
     end
   end
 
-  def team_average_goals(team_id)
-    teams_by_id = @game_teams_array.find_all do |gameteam|
-      gameteam.team_id == team_id
+  def assign_goals_by_team_hash
+      team_goals = Hash.new { |hash, key| hash[key] = [] }
+      @game_teams_array.each do |gameteam|
+        team_goals[gameteam.team_id] = []
+      end
+      @game_teams_array.each do |gameteam|
+        team_goals[gameteam.team_id] << gameteam.goals.to_i
+      end
+      team_goals
+  end
+
+    def average_goals_by_team(team_goals)
+      team_goals.keys.each{|team|
+        team_goals[team] = (team_goals[team].sum.to_f/(team_goals[team].size)).round(2)}
+      team_goals
     end
 
-    total_goals = teams_by_id.sum do |team|
-      team.goals.to_i
+    def teams_max_by_average_goal(team_goals)
```

Comment:
Awesome work John. This makes everything fun wwaayy faster. Thanks for doing that